src/RateTable1.java

```diff
@@ -0,0 +1,127 @@
+import java.math.BigDecimal;
+
+import components.map.Map;
+import components.map.Map.Pair;
+import components.map.Map1L;
+
+/**
+ * {@code RateTable} represented as a {@code Map} with implementations of
+ * primary methods.
+ *
+ * @convention <pre>
+ * [the RateTable map is non-null, contains unique keys (currency names),
+ *             and each associated BigDecimal value represents a valid exchange rate]
+ * </pre>
+ * @correspondence <pre>
+ * this = [map of currency names (String) to exchange rate values (BigDecimal)]
+ * </pre>
+ *
+ * @author Ryan Ouimet
+ *
+ */
+public class RateTable1 extends RateTableSecondary {
+
+    /*
+     * Private members --------------------------------------------------------
+     */
+
+    /**
+     * Representation of {@code this}.
+     */
+    private Map<String, BigDecimal> rep;
+
+    /**
+     * Creator of initial representation.
+     */
+    private void createNewRep() {
+        this.rep = new Map1L<>();
+    }
+
+    /*
+     * Constructors -----------------------------------------------------------
+     */
+
+    /**
+     * No-argument constructor.
+     */
+    public RateTable1() {
+        this.createNewRep();
+    }
+
+    /*
+     * Standard methods -------------------------------------------------------
+     */
+
+    @Override
+    public final RateTable newInstance() {
+        try {
+            return this.getClass().getConstructor().newInstance();
+        } catch (ReflectiveOperationException e) {
+            throw new AssertionError(
+                    "Cannot construct object of type " + this.getClass());
+        }
+    }
+
+    @Override
+    public final void clear() {
+        this.createNewRep();
+    }
+
+    @Override
+    public final void transferFrom(RateTable source) {
+        assert source != null : "Violation of: source is not null";
+        assert source != this : "Violation of: source is not this";
+        assert source instanceof RateTable1 : ""
+                + "Violation of: source is of dynamic type NaturalNumberExample";
+        /*
+         * This cast cannot fail since the assert above would have stopped
+         * execution in that case.
+         */
+        RateTable1 localSource = (RateTable1) source;
+        this.rep = localSource.rep;
+        localSource.createNewRep();
+    }
+
+    /*
+     * Kernel methods ---------------------------------------------------------
+     */
+
+    @Override
+    public final ExchangeRate getExchangeRate(String name) {
+        assert name != null : "Violation of: name is not null";
+        return new ExchangeRate(name, this.rep.value(name));
```

Comment:
Very interesting design here. You avoid aliasing altogether by returning a new object. You may not need to do this because ExchangeRate is immutable, but this is excellent work.

Reply:
I just noticed your map doesn't store exchange rates, so this is completely necessary.